osu.Game.Rulesets.Osu/Skinning/Legacy/LegacyMainCirclePiece.cs

```diff
@@ -22,53 +24,59 @@ public class LegacyMainCirclePiece : CompositeDrawable
     {
         public override bool RemoveCompletedTransforms => false;
 
-        private readonly string priorityLookup;
         private readonly bool hasNumber;
 
-        public LegacyMainCirclePiece(string priorityLookup = null, bool hasNumber = true)
-        {
-            this.priorityLookup = priorityLookup;
-            this.hasNumber = hasNumber;
+        /// <summary>
+        /// A prioritised prefix to perform texture lookups with.
+        /// </summary>
+        /// <remarks>
+        /// If the "circle" texture could not be found with this prefix,
+        /// then it is nullified and the default prefix "hitcircle" is used instead.
+        /// </remarks>
+        private string? priorityLookupPrefix;
 
-            Size = new Vector2(OsuHitObject.OBJECT_RADIUS * 2);
-        }
+        private Drawable hitCircleSprite = null!;
 
-        private Drawable hitCircleSprite;
+        protected Container OverlayLayer { get; private set; } = null!;
 
-        protected Container OverlayLayer { get; private set; }
-
-        private Drawable hitCircleOverlay;
-        private SkinnableSpriteText hitCircleText;
+        private Drawable hitCircleOverlay = null!;
+        private SkinnableSpriteText hitCircleText = null!;
 
         private readonly Bindable<Color4> accentColour = new Bindable<Color4>();
         private readonly IBindable<int> indexInCurrentCombo = new Bindable<int>();
 
-        [Resolved]
-        private DrawableHitObject drawableObject { get; set; }
+        [Resolved(canBeNull: true)]
+        private DrawableHitObject? drawableObject { get; set; }
 
         [Resolved]
-        private ISkinSource skin { get; set; }
+        private ISkinSource skin { get; set; } = null!;
+
+        public LegacyMainCirclePiece(string? priorityLookupPrefix = null, bool hasNumber = true)
+        {
+            this.priorityLookupPrefix = priorityLookupPrefix;
+            this.hasNumber = hasNumber;
+
+            Size = new Vector2(OsuHitObject.OBJECT_RADIUS * 2);
+        }
 
         [BackgroundDependencyLoader]
         private void load()
         {
-            var drawableOsuObject = (DrawableOsuHitObject)drawableObject;
-
-            bool allowFallback = false;
+            var drawableOsuObject = (DrawableOsuHitObject?)drawableObject;
 
             // attempt lookup using priority specification
-            Texture baseTexture = getTextureWithFallback(string.Empty);
+            Texture? baseTexture = getTexture(string.Empty);
 
-            // if the base texture was not found without a fallback, switch on fallback mode and re-perform the lookup.
+            // if the base texture was not found using the priority specification, nullify the specification and fall back to "hitcircle".
             if (baseTexture == null)
             {
-                allowFallback = true;
-                baseTexture = getTextureWithFallback(string.Empty);
+                priorityLookupPrefix = null;
```

Comment:
I'm not sure I like this. I feel like the prefix should not be set to `null` like this. Instead `getTexture` should be called twice with the prefix passed in.

Reply:
Right, this can be vastly simplified...